### PR TITLE
Rc 274 passing sites only

### PR DESCRIPTION
**Pull request by RoriCremer**
Locally tested and the exclude-filtered flag removes filtered rows
Screen Shot 2021-05-25 at 7 09 22 PM

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohort.java [hunk 1]
```diff
@@ -128,6 +128,16 @@
...
 
+    /**
+    * If this flag is enabled, sites that have been marked as filtered (i.e. have anything other than `.` or `PASS`
+    * in the FILTER field) will be excluded from the output.
+    */
+    @Argument(
+            fullName="exclude-filtered",
+            doc="Don't include filtered sites in the final jointVCF",
+            optional=true)
+    private boolean XLfiltered = false;
```
Comment (kcibul) [inline src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohort.java:139]: typically Java variables start with lower case, (and XL looks like extra large to me!).  How about `excludeFilteredSites` just to be super clear?

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortEngine.java [hunk 2]
```diff
@@ -429,7 +433,7 @@ private void finalizeCurrentVariant(final List<VariantContext> unmergedCalls,
 
         // apply VQSLod-based filters
         VariantContext filteredVC =
-                noVqslodFilteringRequested ? genotypedVC : filterSiteByVQSLOD(genotypedVC, vqsLodMap, yngMap, performGenotypeVQSLODFiltering);
+                noVqslodFilteringRequested ? genotypedVC : filterSiteAlleleByVQSLOD(genotypedVC, vqsLodMap, yngMap, performGenotypeVQSLODFiltering);
```
Comment (kcibul) [inline src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortEngine.java:436]: Technically this is filtering at the site level (ie the entire site gets filtered, not a site-allele).  If you wanted to make it more specific, maybe `filterSiteByAlleleSpecificVQSLOD` would be more accurate

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortEngine.java [hunk 3]
```diff
@@ -450,12 +454,15 @@ private void finalizeCurrentVariant(final List<VariantContext> unmergedCalls,
         final VariantContext finalVC = removeAnnotations(filteredVC);
 
         if ( finalVC != null ) {
-            vcfWriter.add(finalVC);
+            // If XLfiltered is true, than the filtered sites should not be added
+            if (!XLfiltered || finalVC.isNotFiltered()) {
```
Comment (kcibul) [inline src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortEngine.java:458]: this made my brain hurt :D.  What do you think of `! (XLfiltered && finalVC.isFiltered)`.  Meh, ok I don't love that either 🤷

**Review by kcibul — APPROVED** [review]
code looks great.  Feels like we could use some better tests involving filtering in general, maybe make a ticket for that?

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortEngine.java (hunk 3 above)
Comment (RoriCremer) [reply src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortEngine.java:458]: hahaha Andrea listened to me repeat this about 30 times before it was correct. I'm not sure of the best option, but I can comment it more clearly for sure

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohort.java (hunk 1 above)
Comment (RoriCremer) [reply src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohort.java:139]: yeah---XL did feel weird---but I wanted to lift it directly from SelectVariants. Is it better to be clear or match the already existing pattern?

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortEngine.java (hunk 3 above)
Comment (RoriCremer) [reply src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortEngine.java:458]: is it any clearer if I just swap them?
if (finalVC.isNotFiltered() || !XLfiltered ) { foo }

**Pull request merged**